orte/runtime/help-orte-runtime.txt

```diff
@@ -61,3 +62,13 @@ as the mapping.
 Open MPI was unable to determine the number of nodes in your allocation. We
 are therefore assuming a very large number to ensure you receive proper error
 messages.
+#
+[failed-to-uncompress]
+A compressed message was received that could not be
+decompressed. This is most likely due to a missing
+libz library on the receiving node:
+
+  node:  %s
+
+Please ensure that the libz library is present on all
+compute nodes.
```

Comment:
```suggestion
A compressed message was received by the Open MPI run time system that
could not be decompressed.  This means that Open MPI includes
compression support on one node, but does not include compression
support on another node.  This is an unsupported configuration.

Specifically: all instances of the Open MPI run time in a single job
must be homogeneous in terms of compression support (i.e., they all
have compression support, or they all do not have compression
support).

  node without compression support:  %s
```

How's this?  The original message says that libz has to be present, but that's not quite accurate.  It's not just libz that has to be present, but Open MPI has to be built with support for compression (e.g., be able to find the libz headers).

Reply:
I have no strong opinion on the message for OMPI. However, your statement isn't quite correct. If OMPI wasn't built with compression support, then you will never see this error as `mpirun` won't compress the message. So if you are getting this error, the only possible reason is that OMPI _was_ built with compression support (i.e., with libz), which means that both `mpirun` and the `orted` were built with libz - and the libz library was available on the node where `mpirun` resides but wasn't available on at least one backend, which caused the compression component on that node to fail to load.

Note that everything is just fine if libz is not available where `mpirun` executes as compression will simply not be done. Doesn't matter what is on the backend in that case.